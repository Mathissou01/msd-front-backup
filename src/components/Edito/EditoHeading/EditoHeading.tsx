import Image from "next/image";
import classNames from "classnames";
import {
  Maybe,
  TagEntity,
  UploadFile,
} from "../../../graphql/codegen/generated-types";
import { makePublicAssetPath } from "../../../lib/utilities";
import MobileActuHero from "public/images/mobile_actuhero.svg";
import DesktopActuHero from "public/images/desktop_actuhero.svg";
import "./edito-heading.scss";

interface IEditoHeadingProps {
  title: string;
  tags?: Array<TagEntity>;
  image?: Maybe<UploadFile>;
}

export default function EditoHeading({
  title,
  tags,
  image,
}: IEditoHeadingProps) {
  const hasImage = !!image;
  const headingContent = classNames("c-EditoHeading__Content", {
    "c-EditoHeading__Content_noImage": !hasImage,
  });
  const cardContainer = classNames("c-EditoHeading__CardContainer", {
    "c-EditoHeading__CardContainer_fullWidth": !hasImage,
  });
  const headingCard = classNames("c-EditoHeading__Card", {
    "c-EditoHeading__Card_only": !hasImage,
  });

  return (
    <div className="c-EditoHeading">
      <div className={headingContent}>
        {image && (
          <div className="c-EditoHeading__Header">
            <Image
              src={makePublicAssetPath(image.url)}
              alt={image.alternativeText ?? ""}
              fill
              priority
            />
          </div>
        )}
        <div className={cardContainer}>
          <div className={headingCard}>
            <h1 className="c-EditoHeading__Title">{title}</h1>
            <div className="c-EditoHeading__Tags">
              {tags?.map((tagLabel, index) => (
                <span
                  key={index}
                  className="c-EditoHeading__Tag c-EditoHeading__Tag_background"
                >
                  {tagLabel.attributes?.name}
                </span>
              ))}
            </div>
          </div>
        </div>
      </div>
      <div className="c-EditoHeading__Background" />
      <div className="c-EditoHeading__SvgContainer">
        <MobileActuHero className="c-EditoHeading__Svg_mobile" />
        <DesktopActuHero className="c-EditoHeading__Svg_desktop" />
      </div>
    </div>
  );
}
