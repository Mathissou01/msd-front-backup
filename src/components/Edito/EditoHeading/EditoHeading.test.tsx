import { render } from "@testing-library/react";
import EditoHeading from "./EditoHeading";

const mock = {
  title: "Edito heading title",
  tagLabels: [
    {
      id: "16",
      attributes: {
        name: "Prise de RDV",
      },
    },
    {
      id: "12",
      attributes: {
        name: "EvÃ©nement",
      },
    },
  ],
  image: {
    url: "/images/images-temp/temp_image.jpg",
    alternativeText: "",
    hash: "",
    mime: "",
    name: "",
    provider: "",
    size: 0,
  },
};

describe("EditoHeading", () => {
  it("renders", () => {
    const { container } = render(
      <EditoHeading
        title={mock.title}
        image={mock.image}
        tags={mock.tagLabels}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
