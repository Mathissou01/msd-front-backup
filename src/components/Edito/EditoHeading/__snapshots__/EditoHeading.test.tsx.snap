// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditoHeading renders 1`] = `
<div>
  <div
    class="c-EditoHeading"
  >
    <div
      class="c-EditoHeading__Content"
    >
      <div
        class="c-EditoHeading__Header"
      >
        <img
          alt=""
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fimages-temp%2Ftemp_image.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="c-EditoHeading__CardContainer"
      >
        <div
          class="c-EditoHeading__Card"
        >
          <h1
            class="c-EditoHeading__Title"
          >
            Edito heading title
          </h1>
          <div
            class="c-EditoHeading__Tags"
          >
            <span
              class="c-EditoHeading__Tag"
            >
              Prise de RDV
            </span>
            <span
              class="c-EditoHeading__Tag c-EditoHeading__Tag_darker"
            >
              Ev√©nement
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c-EditoHeading__Background"
    />
    <div
      class="c-EditoHeading__SvgContainer"
    >
      <div
        class="c-EditoHeading__Svg_mobile"
      />
      <div
        class="c-EditoHeading__Svg_desktop"
      />
    </div>
  </div>
</div>
`;
